app/src/shared/lib/hooks/useInfiniteScroll.ts

```diff
@@ -0,0 +1,55 @@
+import { RefObject, useCallback, useEffect, useRef } from 'react';
+
+/**
+ *
+ * @param rootElement - ссылка на корневой элемент, который необходим для наблюдения за пересечением с последним элементом
+ * @param trigger - триггер, при котором будет происходить перерасчет последнего элемента
+ * @param callback - функция, которая будет вызываться при пересенчении корневого элемента и последнего элемента
+ */
+
+export const useInfiniteScroll = <T extends HTMLElement | null>(
+    rootElement: RefObject<T>,
+    trigger: number | string | boolean,
+    callback: () => void,
+) => {
+    const lastElement = useRef<ChildNode | null | undefined>(null);
+    const observer = useRef<IntersectionObserver | null>(null);
+
+    const handleIntersection = useCallback(
+        (entries: IntersectionObserverEntry[]) => {
+            if (
+                entries[0].isIntersecting &&
+                entries[0].target === lastElement.current
```

Comment:
```suggestion
        ([entry]: IntersectionObserverEntry[]) => {
            if (
                entry.isIntersecting &&
                entry.target === lastElement.current
```
https://learn.javascript.ru/destructuring#massiv